### PR TITLE
Added delete shoe feature

### PR DESCRIPTION
**Pull request by JusLafo**
- Added a column called "is_admin" which is a boolean
- Added a button on the shoeShowPage that only shows when someone has an admin status of true
- Added functionality of a delete button to delete the specified shoe

### CODE REVIEW

File: client/src/components/ShoeShowPage.js [hunk 1]
```diff
@@ -1,41 +1,91 @@
...
+import translateServerErrors from "../services/translateServerErrors"
 
 const ShoeShowPage = props => {
     const [shoe, setShoe] = useState({})
+    const [shouldRedirect, setShouldRedirect] = useState(false)
 
     const id = props.match.params.id
+
     const getShoe = async () => {
         try{
```
Comment (dgazzola) [inline client/src/components/ShoeShowPage.js]: space after try

File: client/src/components/ShoeShowPage.js [hunk 2]
```diff
@@ -1,41 +1,91 @@
...
-                const error = new Error(errorMessage)
-                throw(error)
-              }
+              const errorMessage = `${response.status} (${response.statusText})`
+              const error = new Error(errorMessage)
+              throw(error)
+            }
             const shoeData = await response.json()
             setShoe(shoeData.shoe)
         }   catch(err) {
```
Comment (dgazzola) [inline client/src/components/ShoeShowPage.js]: theres a double space before catch

File: client/src/components/ShoeShowPage.js [hunk 3]
```diff
@@ -1,41 +1,91 @@
...
+          const body = await response.json()
+          setShouldRedirect(true)
+        }
+      } catch (err) {
+        console.error(`Error in fetch: ${err.message}`)
+      }
+    }
 
     useEffect(() => {
         getShoe()
```
Comment (dgazzola) [inline client/src/components/ShoeShowPage.js]: not 100% but this might be double indented

File: client/src/components/ShoeShowPage.js [hunk 4]
```diff
@@ -1,41 +1,91 @@
...
+    if (shouldRedirect) {
+      return <Redirect push to="/" />
+    }
+
+    let deleteButton = ''
+    if (props.user?.is_admin == true) {
+      deleteButton = <button type="button" className="button" onClick={handleDelete}>Delete Shoe</button>
+    }
 
     return(
```
Comment (dgazzola) [inline client/src/components/ShoeShowPage.js]: think space before parenths

File: client/src/components/UserShowPage.js [hunk 5]
```diff
@@ -26,49 +27,54 @@ const UserShowPage = (props) => {
...
-            if(!response.ok) {
-              const errorMessage = `${response.status}: (${response.statusText})`
-              throw new Error(errorMessage)
-            }
-            const body = await response.json()
-            setUser(body.user)
-          } catch (error) {
-            console.error(`error in fetch: ${error}`)
+        const response = await fetch(`/api/v1/users/${id}`)
+          if(!response.ok) {
```
Comment (dgazzola) [inline client/src/components/UserShowPage.js]: space after if

File: client/src/components/UserShowPage.js [hunk 6]
```diff
@@ -100,6 +106,7 @@ const UserShowPage = (props) => {
         <div>
           <div>
               <h1>{user.username}'s Profile</h1>
```
Comment (dgazzola) [inline client/src/components/UserShowPage.js]: I think this chunk of child elements is double indented

**Review by dgazzola — COMMENTED** [review]
looks dope, just a couple spacing things

File: client/src/components/App.js [hunk 7]
```diff
@@ -34,10 +34,9 @@ const App = (props) => {
         <Route exact path="/" component={ShoesListPage} />
         <Route exact path="/shoes" component={ShoesListPage} />
         <Route exact path="/shoes/new" component={NewShoeForm} />
-        <Route 
-          exact 
-          path="/shoes/:id" 
-          render={props => <ShoeShowPage {...props} user={currentUser}/>}
+        <Route exact path="/shoes/:id"
+          render= {
+            (props) => <ShoeShowPage {...props} user={currentUser}/>}
```
Comment (nwalberts) [inline client/src/components/App.js:39]: indentation is off. 
```
<Route 
     props
/>
```

File: client/src/components/ShoeShowPage.js [hunk 8]
```diff
@@ -1,92 +1,95 @@
...
-      reviewFormComponent = <ReviewForm postReview={postReview} shoe={shoe} errors={errors} setErrors={setErrors}/>
+  useEffect(() => {
+    getShoe()
+  }, [])
+
+  const handleDelete = () => {
+    if (confirm("Are you sure you want to delete this shoe? \nThis cannot be undone") == true) {
+      deleteShoe()
+    } else {
+      alert("Delete Cancelled")
```
Comment (nwalberts) [inline client/src/components/ShoeShowPage.js]: I would probably say "Your shoe has not been deleted."

File: client/src/components/ShoeShowPage.js [hunk 9]
```diff
@@ -1,92 +1,95 @@
...
+      <div>
+        <h5>Category: {shoe.category}</h5>
+        <h5>Color: {shoe.color}</h5>
+        <h6>Description:</h6>
+        <p>
+          {shoe.description}
+        </p>
+        <div>
+          {deleteButton}
         </div>
```
Comment (nwalberts) [inline client/src/components/ShoeShowPage.js:89]: this p tag can be written on one line
this div can be written on one line

File: client/src/components/ShoeShowPage.js [hunk 10]
```diff
@@ -1,92 +1,95 @@
...
-            </p>
-            <ReviewList reviews={shoe.reviews}/>
-            <div>
-              {reviewFormComponent}
-            </div>
+  return (
+    <div className = "callout">
+      <h1>{shoe.name}</h1>
+      <img src={shoe.url} />
+      <div>
```
Comment (nwalberts) [inline client/src/components/ShoeShowPage.js:80]: this div isnt doing anything, lets delete it

File: client/src/components/UserShowPage.js [hunk 11]
```diff
@@ -2,110 +2,119 @@ import React, { useState } from 'react';
...
-    })
-    const [newProfileImage, setNewProfileImage] = useState({
-      image: {}
-    })
+  const { id } = props.match.params
+  const currentUser = props.currentUser
+
+  const [user, setUser] = useState({
+    id: '', email: '', createdAt: '', updatedAt: '', username: '', profileImage: ''
+  })
```
Comment (nwalberts) [inline client/src/components/UserShowPage.js:10]: probably dont need all of these properties on this default state, and it looks a bit wild. remove

File: client/src/components/UserShowPage.js [hunk 12]
```diff
@@ -2,110 +2,119 @@ import React, { useState } from 'react';
...
-        }, [])
+  let adminText = ""
+  if (currentUser?.is_admin) {
+    adminText = <h6>Signed in as administrator</h6>
+  }
+    
+  let dropzoneComponent = ""
+  if(currentUser?.id === user.id){
+    dropzoneComponent = (
+      <div>
```
Comment (nwalberts) [inline client/src/components/UserShowPage.js:82]: this div isnt doing anything. remove it and/or replace it with a React fragment

File: client/src/components/UserShowPage.js [hunk 13]
```diff
@@ -2,110 +2,119 @@ import React, { useState } from 'react';
...
-              <h4>{user.email}</h4>
-              <p>This user has been user since {createdDateString}</p>
-              <img src={user.profileImage} className='profile-image' alt='profile-image' />
-          </div>
-          {dropzoneComponent}
-        </div>
-    );
+  return ( 
+    <div>
+      <div>
```
Comment (nwalberts) [inline client/src/components/UserShowPage.js]: both of these divs arent doing anything. replace the outer one with a react fragment, and nuke the inner one

File: server/src/routes/api/v1/shoesRouter.js [hunk 14]
```diff
@@ -6,6 +6,15 @@ import shoeReviewsRouter from "./shoeReviewsRouter.js"
 
 const shoesRouter = new express.Router()
 
+shoesRouter.delete("/:id", async (req, res) => {
+  const { body } = req
+  try {
+    const deletedShoe = await Shoe.query().findOne("name", body.name).delete()
```
Comment (nwalberts) [inline server/src/routes/api/v1/shoesRouter.js:12]: lets use `deleteById` instead. its faster!

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Pull request merged**